text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/Inserting.java

```diff
@@ -21,41 +21,35 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-package net.kyori.adventure.text.minimessage.transformation;
+package net.kyori.adventure.text.minimessage.tag;
 
-import net.kyori.adventure.internal.Internals;
 import net.kyori.adventure.text.Component;
-import net.kyori.examination.Examinable;
 
 /**
- * A transformation that can be applied while parsing a message.
+ * A tag type that inserts a {@link Component} into the output.
  *
- * <p>A transformation instance is created for each instance of a tag in a parsed string.</p>
- *
- * @see TransformationRegistry to access and register available transformations
  * @since 4.10.0
  */
-public abstract class Transformation implements Examinable {
-
-  protected Transformation() {
-  }
-
+public /* non-sealed */ interface Inserting extends Tag {
```

Comment:
yes, they should be, it should be API, for people to build their own tags easily

Reply:
Well, there are static constructor methods on `Tag` that create instances of these tags already. Although I suppose this could work for lazy insertions? Although maybe we should just add a supplier insertion?

Reply:
I'd intended for end-users to be allowed to implement their own versions of `Inserting`/`PreProcess`/`Modifying` for any more complex logic, with the factory methods existing as shorthand for simple cases.